[2023.2][Backport] [UUM-53413] Fix GC memory heap reporting

bandures opened the pull request:
This PR is to update BDWGC repo with fixes to heap reporting functions.

Minor change:
- Fixes build script to detect different versions of Visual Studio properly

**Release notes**

Fixed UUM-53413@username:
Mono: Fix GC heap reporting to report reserved (free) sections

https://github.com/Unity-Technologies/bdwgc/pull/86

Commit:
[UUM-53413] Fix GC memory heap reporting

joncham commented:
> This PR is to update BDWGC repo with fixes to heap reporting functions.

I don't see a change to bdwgc here?

Commit:
Address PR feedback

bandures commented:
> > This PR is to update BDWGC repo with fixes to heap reporting functions.
> 
> I don't see a change to bdwgc here?

nice catch, added a wrong item.
fixed
